KIPs/kip-draft_kgp6.md

```diff
@@ -0,0 +1,652 @@
+                for (uint256 j = 0; j < approvers.length; j++) {
+                    _validateAdmin(retirees[i].retired, approvers[j]);
+                    minApprovals++;
+                }
+                require(
+                    minApprovals >= req,
+                    "min required admins should approve"
+                );
+            } else {
+                revert("min required admins should approve");
+            }
+        }
+    }
+
+    /**
+     * @dev sets the status of the contract to Finalize. Once finalized the storage data
+     * of the contract cannot be modified
+     * @param _memo is the result of the rebalance after executing successfully in the core.
+     */
+    function finalizeContract(
+        string memory _memo
+    ) public onlyOwner onlyAtStatus(Status.Approved) {
+        memo = _memo;
+        status = Status.Finalized;
+        emit Finalized(memo, status);
+    }
+
+    /**
+     * @dev resets all storage values to empty objects except targetBlockNumber
+     */
+    function reset() public onlyOwner {
+        //reset cannot be called at Finalized status or after target block.number
+        require(
+            ((status != Status.Finalized) &&
+                (block.number < rebalanceBlockNumber)),
+            "Contract is finalized, cannot reset values"
+        );
+
+        //`delete` keyword is used to set a storage variable or a dynamic array to its default value.
+        delete retirees;
+        delete newbies;
+        delete memo;
+        status = Status.Initialized;
+    }
+
+    //Getters
+    /**
+     * @dev to get retired details by retiredAddress
+     * @param _retiredAddress is the address of the retired
+     */
+    function getRetired(
+        address _retiredAddress
+    ) public view returns (address, address[] memory) {
+        require(retiredExists(_retiredAddress), "Retired does not exist");
+        uint256 index = getRetiredIndex(_retiredAddress);
+        Retired memory retired = retirees[index];
+        return (retired.retired, retired.approvers);
+    }
+
+    /**
+     * @dev check whether retiredAddress is registered
+     * @param _retiredAddress is the address of the retired
+     */
+    function retiredExists(address _retiredAddress) public view returns (bool) {
+        require(_retiredAddress != address(0), "Invalid address");
+        for (uint8 i = 0; i < retirees.length; i++) {
+            if (retirees[i].retired == _retiredAddress) {
+                return true;
+            }
+        }
+    }
+
+    /**
+     * @dev get index of the retired in the retirees array
+     * @param _retiredAddress is the address of the retired
+     */
+    function getRetiredIndex(
+        address _retiredAddress
+    ) public view returns (uint256) {
+        for (uint256 i = 0; i < retirees.length; i++) {
+            if (retirees[i].retired == _retiredAddress) {
+                return i;
+            }
+        }
+        revert("Retired does not exist");
```

Comment:
Since this is a view function, we don't need to have revert here. return something is enough like -1 (UINT256_MAX)

Reply:
fixed